backend/src/Service/OAICompletionService.php

```diff
@@ -100,4 +101,52 @@ public function countTextsTokens(array $messages): int
 
         return $tokens;
     }
+
+    private function parseMultiToolResponse(CreateResponse $response): CreateResponse
```

Comment:
While parsing the multi-tool response, ensure that the transformation of the response back into a `CreateResponse` object maintains all necessary attributes and does not unintentionally alter the expected outcome of the original response.